web-src/src/services/FirefallService.js

```diff
@@ -9,9 +9,46 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+
 import { wretch } from '../helpers/NetworkHelper.js';
 import { replaceRuntimeDomainInUrl } from '../helpers/UrlHelper.js';
 
+export const FIREFALL_ACTION_TYPES = {
+  TEXT_TO_IMAGE_PROMPT_GENERATION: 'text-to-image',
+  VARIATIONS_GENERATION: 'variations',
+};
+
+const FIREFALL_POLLING_TIMEOUTS = {
```

Comment:
It looks like an attempt to simulate enums in JS. It makes the code bulkier, so I'm curious if there are any practical benefits to using this technique. Since objects aren't frozen, the properties are mutable, so this structure doesn't really behave like an enum. Could extracting the field into a constant make it safer and also less verbose?

Reply:
It is an _internal_ const just for grouping related things. That is not needed, and I can use the individual constants instead.
>Could extracting the field into a constant make it safer and also less verbose?

yes